cmds/live.c

```diff
@@ -238,6 +238,86 @@ static int query_agent_capabilities(int fd)
 	return status;
 }
 
+/**
+ * forward_option - forward a given option and its data to the agent
+ * @fd           - agent socket file descriptor
+ * @capabilities - features supported by the agent
+ * @opt          - option type to send
+ * @value        - option data load
+ * @value_size   - size of @value
+ * @return       - status: -1 on error, 0 on success
+ */
+static int forward_option(int fd, int capabilities, int opt, void *value, size_t value_size)
+{
+	void *data;
+	size_t data_size;
+	struct uftrace_msg ack;
+	int ret = -1;
+
+	if (!(opt & capabilities)) {
+		pr_dbg2("skip unsupported option\n");
```

Comment:
Don't we need to warn the user about the unsupported options?

Reply:
I'm note sure when to use `pr_warn()`, maybe you can a word about it in the wiki? Is it for major events, or unexpected states that the user should have a look at? I see it as an error prevents normal execution, but which is not fatal to the program. And that is not the case if we're skipping an option.

What about other feedback? The client command is not interactive so I don't expect it to communicate with the user outside of the debug messages, which the user has to ask for. For example if the agent fails to apply a filter, should we warn the user too? Or just log it in debug messages so the user can check?

Reply:
Basically it should call `pr_warn()` if it fails to do something user requested.  It might be a debug message if there's a way to work around it.  I think the client command just forwards the option to the agent, then failing on applying a filter would need `pr_warn()`.